### PR TITLE
PR: pep8, pylint, code style and general refactor of fsm application

### PR DESCRIPTION
**Pull request by dixon-che**
@cjlee112 

- quality.sh: implement new syntax `./check-quality.sh {pep8|pylint} {lti|psa|fsm|ct|mysite|all}` to allow to check `pep8|pylint` issues separately for all applications
- mysite/fsm/models.py: add `JSONBlobMixin` to move project to django/python best practices of making a readable and supportable code
- mysite/fsm/utils.py: move `get_plugin` func to utils.py to separate logic that not belong to `django models`
- README.rst: update according to new check-quality syntax
- all `fsm` app: change code according to pep8 and our code stype. Resolve important pylint warnings.
- remove unnecessary get_or_create method from ActivityLog

Issue https://github.com/cjlee112/socraticqs2/issues/52
Task on [trello](https://trello.com/c/h6K1UkxM/168-fsm-refactor-pep8-pylint-add-jsonblobmixin).

### CODE REVIEW

File: mysite/fsm/models.py [hunk 1]
```diff
@@ -82,78 +41,106 @@ def save_json_data(self, d=None, attr='data', doSave=True):
 )
 
 
-def load_json_id(name, pk):
+class JSONBlobMixin(object):
```
Comment (cjlee112) [inline mysite/fsm/models.py]: @maxsocl I think the only methods that belong in this mixin are the four that actually *are* methods: load_json_data(), save_json_data(), get_data_attr(), and set_data_attr().  The others do not even use self or cls, and should be left as functions: dump_json_id(), dump_json_id_dict(), load_json_id(), load_json_id_dict().

File: mysite/fsm/tests.py [hunk 2]
```diff
@@ -6,10 +6,7 @@
     FSMNode,
     FSMState,
     ActivityLog,
-    dump_json_id,
```
Comment (cjlee112) [inline mysite/fsm/tests.py]: @maxsocl These functions have no reason to be in the mixin class.

File: mysite/fsm/models.py (hunk 1 above)
Comment (maxsocl) [inline mysite/fsm/models.py]: We decided not to change code at this point and to make a note in issue tracker and possible using this function in future in general way.

File: mysite/fsm/tests.py (hunk 2 above)
Comment (maxsocl) [inline mysite/fsm/tests.py]: Same as above comment.

**Pull request merged**